git-keeper-core/gkeepcore/student.py

```diff
@@ -143,7 +143,10 @@ def students_from_csv(reader: CSVReader) -> list:
 
     for row in reader.get_rows():
         if len(row) > 0:
-            students.append(Student.from_csv_row(row))
+            if Student.from_csv_row(row) in students:
+                raise StudentError(Student.email_address + ' appears more than once.')
+            else:
+                students.append(Student.from_csv_row(row))
```

Comment:
`Student.from_csv_row(row)` parses the CSV row and returns a `Student` object. Rather than calling this method twice, it would be better to call it once and store the returned object. Something like this:

```
student = Student.from_csv_row(row)
```

Then you can simply say `if student in students` and `students.append(student)`. This also gives you an object from which to get the email address. `Student.email_address` does not work, because `Student` is a class, not an object, and `email_address` is an attribute at the object level.

It is also better to use string formatting rather than adding strings, like this:

```
'{} appears more than once'.format(student.email_address)
```